mesonbuild/build.py

```diff
@@ -1987,6 +1987,9 @@ def post_init(self) -> None:
             # Don't let configuration proceed with a non-static crate type
             elif self.rust_crate_type not in ['rlib', 'staticlib']:
                 raise InvalidArguments(f'Crate type "{self.rust_crate_type}" invalid for static libraries; must be "rlib" or "staticlib"')
+            # See https://github.com/rust-lang/rust/issues/110460
+            if self.rust_crate_type != 'rlib' and any(c in self.name for c in ['-', ' ']):
```

Comment:
```suggestion
            if self.rust_crate_type != 'rlib' and any(c in self.name for c in {'-', ' '}):
```



nitpicking: python syntax for a set is {'-', ' '}, which is supposedly faster than a list, not that it matters much for a 2 items set. Same below.

Reply:
Faster for membership checks, specifically. :) Not for-loops.